notifications_utils/recipients.py

```diff
@@ -186,36 +182,17 @@ def get_rows(self):
                 for key in column_headers[length_of_row:]:
                     insert_or_append_to_dict(output_dict, key, None)
 
-            yield Columns(output_dict)
-
-    @property
-    def rows_with_errors(self):
-        return self.rows_with_missing_data | self.rows_with_bad_recipients | self.rows_with_message_too_long
-
-    @property
-    def rows_with_missing_data(self):
-        return set(
-            row['index'] for row in self.annotated_rows if any(
-                value.get('error') == self.missing_field_error
-                for key, value in row['columns'].items()
-            )
-        )
-
-    @property
-    def rows_with_bad_recipients(self):
-        return set(
-            row['index'] for row in self.annotated_rows if any(
-                row['columns'].get(recipient_column, {}).get('error')
-                not in [None, self.missing_field_error]
-                for recipient_column in self.recipient_column_headers
-            )
-        )
-
-    @property
-    def rows_with_message_too_long(self):
-        return set(
-            row['index'] for row in self.annotated_rows if row['message_too_long']
-        )
+            if index < self.max_rows:
```

Comment:
should this if check should be at the top of the for loop to avoid parsing columns at all if there are too many rows

Reply:
Trouble is we don’t know how many rows there are until we’ve processed them…